### ISSUE TITLE
tmp directory is different but it should be the same

### ISSUE THREAD

**Issue by bucaran** [ember-cli/ember-cli#4567]
I built the same ember project in two MBPs with the same configuration and exactly same project under the same conditions, minus one detail. One is SSD, while the other HDD. 

Now, this issue is not about which one is faster, but how is it possible that the output of their respective `tmp` directories is drastically different?

See Issue 
+ https://github.com/broccolijs/broccoli-funnel/issues/27#issuecomment-126634069

The tl;dr is that I suspect there is a bug? somewhere Broc or ember-cli that is causing this misbehavior. So, while having a SSD certainly speeds things up, my data suggests ember-cli or broccoli? is doing something quite different in both machines. But why? and how does it know?

**Comment by stefanpenner** [comment]
Funky. Can you:

* diff both node_modules to ensure no difference?
* provide a simple project that demonstrates the issue. I'll try to track down a similar machine that has an hdd.

My guess if the two trees are the same, some timing issues for tree cleanup. Or some part of the algo is unstable.

If it is unstable it may be related to the DAG of addons we topsort.

**Comment by stefanpenner** [comment]
ping ^

**Comment by bucaran** [comment]
@stefanpenner I was indeed able to confirm the two trees are *always* different when comparing SSD vs HDD machines, but I ran out of time to come up with a blank project to replicate the issue. I am currently migrating their build system to a different stack. The 2~3 min build times was killing their productivity and one dude had already left when I joined.

:v:

**Comment by bucaran** [comment]
The problem has not changed though. The two trees are always different, but it's OK, you can run away from the problem :smile:

**Comment by stefanpenner** [comment]
> The problem has not changed though.

It is not clear this is actually causing a problem, beyond leaking some extra tmp files. Which should be handled by the new broccoli-plugin base class that moves tmp to the system temp, where these files will be reaped naturally.

If some future problem indicates this is not true, it will be evaluated.

> but it's OK, you can run away from the problem

I really don't appreciate your tone, each interaction has been entitled and confrontational. Ultimately providing very little value or insight, but consuming a disproportionate amount of my time.

Anyways, I wish you the best of luck.

**Comment by bucaran** [comment]
Well, you went ahead and closed the issue without further comment. :-1: 

Anyway, thanks.
